src/abstraction/transient.rs

```diff
@@ -217,18 +219,19 @@ impl TransientKeyContext {
     pub fn sign(
         &mut self,
         key_context: TpmsContext,
-        key_auth: &[u8],
-        digest: &[u8],
+        key_auth: Option<Auth>,
+        digest: Digest,
```

Comment:
I like the use of the newtypes, but could we maybe keep the references? As in use `Option<&Auth>` and `&Digest`? That makes the operation easier to use, more efficient (less cloning) and let the potential memory-zeroing responsibality of those buffers to the caller.

Reply:
(similarly anywhere a reference was converted to ownership)

Reply:
Actually, I am now thinking if using `Option<&[u8]>` and `&[u8]` is not better for those function signatures. Those types are nice to use internally in our crate, but should we force the user to use them? With the byte slices, they will be able to call those operations with any type that can be dereferenced to `&[u8]` (like we can do for those types by implementing `Deref`).

Are there any advantages here to force those specific types? Are they TPM specific?

Reply:
I think the advantage of forcing specific types here is that it becomes very clear what kind of data that is expected. And in those types there are also restrictions regarding the data. If we do not use the types then those restrictions would need to be checked every where. I realize that it can be a little bit overhead in the programming by doing this but I think it makes the code clear and harder to get wrong.

Reply:
Are they TPM specific? Well sort of. They are our representation of the TPM2B_DIGEST and TPM2B_AUTH.

Reply:
Agree that the types are good for safety checks and I am ok by keeping them.

What about the the ownership? I'd say it is more friendly to expect a reference instead of be given ownership, if the data are read-only, so that the caller does not have to clone if they want to use the same data after the call.

Reply:
Yes, I agree with you. But I cannot really make up my mind i it should be &Optional<..> or Optional<&..> what do you think makes most sense and is easier to use?

Reply:
I think `Option<&...>` makes more sense (as it is the reference which is optional) and is easier for clients to create (I don't actually know how they would create a `&Option<...>`.
We had similar questions for the [Parsec Rust Client](https://docs.rs/parsec-client/0.6.0/parsec_client/core/basic_client/struct.BasicClient.html#method.psa_asymmetric_encrypt), and we went with `Option<&...>`.

Reply:
Done